crates/weaver_live_check/src/otlp_logger.rs

```diff
@@ -0,0 +1,497 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! OTLP logger provider for emitting policy findings as log records.
+
+use opentelemetry::logs::AnyValue;
+use opentelemetry::logs::LogRecord as _;
+use opentelemetry::logs::LoggerProvider;
+use opentelemetry::logs::{Logger, Severity};
+use opentelemetry::Key;
+use opentelemetry_otlp::WithExportConfig;
+use opentelemetry_sdk::logs::SdkLoggerProvider;
+use opentelemetry_sdk::Resource;
+use serde_json::Value as JsonValue;
+use weaver_checker::{FindingLevel, PolicyFinding};
+
+use crate::{Error, SampleRef};
+
+/// Type alias for log attributes as (Key, AnyValue) pairs
+type LogAttribute = (Key, AnyValue);
+
+/// The service name for weaver resources
+const WEAVER_SERVICE_NAME: &str = "weaver";
+
+/// OTLP emitter for policy findings
+pub struct OtlpEmitter {
+    provider: SdkLoggerProvider,
+}
+
+impl OtlpEmitter {
+    /// Create a new OTLP emitter with gRPC export
+    ///
+    /// NOTE: This must be called from within an active Tokio runtime context
+    /// because the batch exporter spawns background tasks.
+    pub fn new_grpc(endpoint: &str) -> Result<Self, Error> {
+        let exporter = opentelemetry_otlp::LogExporter::builder()
+            .with_tonic()
+            .with_endpoint(endpoint)
+            .build()
+            .map_err(|e| Error::OutputError {
+                error: format!("Failed to create OTLP log exporter: {}", e),
+            })?;
+
+        let provider = SdkLoggerProvider::builder()
+            .with_resource(
+                Resource::builder()
+                    .with_service_name(WEAVER_SERVICE_NAME)
```

Comment:
We should optionally use the OTEL_SERVICE_NAME env variable if it exists.

Reply:
fixed in [cf761cc](https://github.com/open-telemetry/weaver/pull/1045/commits/cf761cc6f17b3aeb84bd6e0b289f7113e772c18b)